tests/test_tags.py

```diff
@@ -391,3 +394,65 @@ def test_copy_related_resource_tag_multi_ref(self):
 
         self.assertEqual(len(untagged_snaps), 1)
         self.assertTrue('Tags' not in untagged_snaps[0].keys())
+
+
+class TagDelayedActionTest(BaseTest):
+
+    def manager(self):
+        return Mock(action_registry={
+            'stop': 0, 'terminate':1, # only care about the keys
+        })
+
+    def action(self, data):
+        return TagDelayedAction(data, self.manager()).validate()
+
+    def test_validate_requires_op(self):
+        try:
+            action = self.action({
+                'type': 'mark-for-op',
+            })
+            self.fail("op should be required")
+        except PolicyValidationError:
+            pass
+        except AssertionError:
+            # Let that fall through
+            raise
+        except Exception as e: # Any other error
+            self.fail(f"unexpected exception: {e}")
```

Comment:
You can streamline this a bit with [`pytest.raises`](https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions)

```suggestion
        with pytest.raises(PolicyValidationError) as exec:
            action = self.action({
                'type': 'mark-for-op',
            })
        assert 'invalid op' in str(exec.value)
```

Reply:
The assert in bit is a bit extra, but just validates we're getting the `PolicyValidationError` we expected

Reply:
How is the exec variable accessible outside the context manager?

Reply:
its accessible/defined outside of the context manager scope